apps/web/app/(app)/simple/SimpleList.tsx

```diff
@@ -47,28 +47,33 @@ import { HtmlEmail } from "@/components/email-list/EmailPanel";
 import { ViewMoreButton } from "@/app/(app)/simple/ViewMoreButton";
 
 export function SimpleList(props: {
-  messages: ParsedMessage[];
+  initialMessages: ParsedMessage[];
   nextPageToken?: string | null;
   userEmail: string;
   type: string;
 }) {
   const { toHandleLater, onSetHandled, onSetToHandleLater } =
     useSimpleProgress();
 
+  const [messages, setMessages] = useState(props.initialMessages);
   const router = useRouter();
 
   const [parent] = useAutoAnimate();
 
   const [isPending, startTransition] = useTransition();
 
-  const toArchive = props.messages
+  const toArchive = messages
     .filter((m) => !toHandleLater[m.id])
     .map((m) => m.threadId);
 
+  const handleUnsubscribe = (id) => {
+    setMessages(messages.filter((message) => message.id !== id));
+  };
+
   return (
     <>
       <div className="mt-8 grid gap-4" ref={parent}>
-        {props.messages
+        {messages
           .filter((m) => !toHandleLater[m.id])
```

Comment:
And then here you can filter by unsubbed `!toHandleLater[m.id] && !unsubscribed[m.id]`